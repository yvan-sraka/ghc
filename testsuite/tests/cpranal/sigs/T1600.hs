-- | Basically tests Nested CPR on IO.
module T1600 where

facIO :: Int -> IO Int
facIO n | n < 2     = return 1
        | otherwise = do n' <- facIO (n-1); return (n*n')
