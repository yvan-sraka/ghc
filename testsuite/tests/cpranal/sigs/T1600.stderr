
==================== Cpr signatures ====================
T1600.$trModule: *
T1600.facIO: *c1(*, #c1(#))


